mlir/include/mlir/IR/DialectImplementation.h

```diff
@@ -112,6 +130,9 @@ struct FieldParser<
     std::enable_if_t<std::is_base_of<Attribute, AttributeT>::value,
                      std::optional<AttributeT>>> {
   static FailureOr<std::optional<AttributeT>> parse(AsmParser &parser) {
+    if constexpr (HasStaticDialectName<AttributeT>::value) {
+      parser.getContext()->getOrLoadDialect(AttributeT::dialectName);
```

Comment:
I wonder if it would be possible to get the TypeID of the attribute class (there should be a static method) and somehow use it to find the dialect instead of relying on the textual name. Not sure if it is possible, I suspect we populate attribute TypeID maps when loading the dialect and here the dialect hasn't been loaded yet...

Reply:
AFAICT TypeID is not static on `Attribute`, nor part of the `tbl-gen` generated code.

Reply:
The way you get TypeID is through `TypeID::get<DialectClass>()`.
I would think that if you can generate a `dialectName` field, you could generate a TypeID one instead?

Reply:
If I had a generated TypeID, I'm still not sure how I would use it to get the dialect. [This method](https://github.com/llvm/llvm-project/blob/3ed76b45be185469f0ff58292102a4202902679d/mlir/include/mlir/IR/MLIRContext.h#L234) seems to still need the dialect namespace as a string.